### DISCUSSION TITLE
How to auto Create a repo from repo?

### DISCUSSION THREAD

**Discussion by piti2** [community/community#38139] [New to GitHub]
Hi there,

I use hugo and nextjs to build websites, now i want that the mairepo to produce another repo and automatically run the build command  to geenrate website files. i currently do it locally as i have 2 repos for each project for example:

- github/username/mynextjs-repo
- github/username/mynextjs-website-repo(this is the repo which holds all the files ready to be pushed live)

Is it possible to have a  nextjs or hugo project which when updated run the build command and update the files on the secondary?

How do you solve this issue, what i want to achieve is to share the files which run a website but do not want to share the project which holds all the data i use to build that website.

Thanks

**Comment by david-engelmann** [comment]
It's possible to be setup as a github action workflow. [This previous discussion should be able to guide you](https://github.com/orgs/community/discussions/26615)

**Reply by piti2** [reply]
Thanks @david-engelmann 

This is very complex for me to make sense of :p right now on local server i run the command "hugo" which build the hugo website. what i need to do to make my master github repo when updated to run hugo and then push the build files to an repo which is connected with cloudflare pages.

Thanks

**Comment by airtower-luna** [comment]
Why do you push the generated website files to a separate repository? That *used to be* a somewhat common approach for using static site generators other than Jekyll, but nowadays you can [set up an Actions workflow](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#publishing-with-a-custom-github-actions-workflow) to publish the generated site directly, no push needed.

Of course you might have another reason, but if you just want to deploy your site that'll be much easier. :slightly_smiling_face:

**Reply by piti2** [reply]
Hi @airtower-luna 

I push the statc files to a seperate repository from where some security options are applied for example the hash, security headers. Since the generated files are managed by 3rd party and i do not wish to share access to the actual project it is much better to have the main project and website files seperately

Currently no static site generator supports auto hashing and security headers generation i have to do it manually. And instead of me doing it manually running local build command and then update the exported repo. i wonder if there is an easier way to auto build wesbtie files on a repo using project of another repo.

Thanks

**Reply by airtower-luna** [reply]
> And instead of me doing it manually running local build command and then update the exported repo.

How about you include the command you would use for the update locally in the build?

Otherwise you'll have to do as @david-engelmann suggested and commit the files to the second repository as part of your workflow. You'll need a PAT in a [secret](https://docs.github.com/en/actions/security-guides/encrypted-secrets) to authorize the access. See the [`actions/checkout` docs](https://github.com/actions/checkout) for how to use the token, check out multiple repositories, commit, and push in a workflow.
